etna/pipeline/hierarchical_pipeline.py

```diff
@@ -61,18 +59,23 @@ def fit(self, ts: TSDataset) -> "HierarchicalPipeline":
         Parameters
         ----------
         ts:
-            Dataset with hierarchical timeseries data
+            Dataset with hierarchical timeseries data.
+        save_ts:
+            Will ``ts`` be saved in the pipeline during ``fit``.
 
         Returns
         -------
         :
             Fitted HierarchicalPipeline instance
         """
-        self._fit_ts = deepcopy(ts)
+        if save_ts:
+            self.ts = ts
```

Comment:
But `ts` is aggregated here, so it is not the same as `self._fit_ts`, isn't it?
```
ts = self.reconciliator.aggregate(ts=ts)
```

Reply:
It isn't the same, but we don't really need self.ts that was before. We only needed `self._fit_ts`, so I think it is better to save `self._fit_ts` in `self.ts`.

Reply:
I don't get it, in the previous line you save copy of `ts` in `self._fit_ts` and now you have modified `ts` in `self.ts`